tests/hap/test_svm_demo.py

```diff
@@ -0,0 +1,146 @@
+""" This module tests full pipeline SVM processing - visit has one detector and uses one filter.
+
+"""
+import datetime
+import glob
+import os
+import pytest
+
+from drizzlepac.haputils import astroquery_utils as aqutils
+from drizzlepac import runsinglehap
+from astropy.io import fits, ascii
+from pathlib import Path
+
+"""
+    test_svm_demo.py
+
+    This test file can be executed in the following manner:
+        $ pytest -s --basetemp=/internal/hladata/yourUniqueDirectoryHere test_svm.py >& test_svm.log &
+        $ tail -f test_svm.log
+      * Note: When running this test, the `--basetemp` directory should be set to a unique
+        existing directory to avoid deleting previous test output.
+      * The POLLER_FILE exists in the tests/hap directory.
+
+    *** The --basetemp does not seem to be working at this time!!!
+
+"""
+
+WCS_SUB_NAME = "FIT_SVM_GAIA"
+POLLER_FILE = "acs_8f6_61_input.out"
+
+@pytest.fixture(scope="module")
+def read_csv_for_filenames():
+    # Read the CSV poller file residing in the tests directory to extract the individual visit FLT/FLC filenames
+    path = os.path.join(os.path.dirname(__file__), POLLER_FILE)
+    table = ascii.read(path, format="no_header")
+    filename_column = table.colnames[0]
+    filenames = list(table[filename_column])
+    print("\nread_csv_for_filenames. Filesnames from poller: {}".format(filenames))
+
+    return filenames
+
+
+@pytest.fixture(scope="module")
+def gather_data_for_processing(read_csv_for_filenames):
+    # Establish FLC/FLT lists and obtain the requested data 
+    flc_list = []
+    flt_list = []
+    flcfiles = []
+    fltfiles = []
+    for fn in read_csv_for_filenames:
+        if fn.lower().endswith("flc.fits"):
+            flc_list.append(fn.split("_")[0])
+        elif fn.lower().endswith("flt.fits"):
+            flt_list.append(fn.split("_")[0])
+
+    # Get test data through astroquery - only retrieve the pipeline processed FLC/FLT files
+    # (e.g., j*_flc.fits).  Soon the SVM and MVM products will be pipeline produced!
+    if flc_list:
+        flcfiles = aqutils.retrieve_observation(flc_list, suffix=["FLC"], product_type="pipeline")
+    if flt_list:
+        fltfiles = aqutils.retrieve_observation(flt_list, suffix=["FLT"], product_type="pipeline")
+
+    flcfiles.extend(fltfiles)
```

Comment:
The function 'retrieve_observation' supports specifying multiple suffix values in the same call, even if there are no products with a specified suffix.  A simple list of rootnames in 'ipppssoot' format with 'suffix=["FLT","FLC"]' provided to a single call to 'retrieve_observation' would replicate this entire function if I am not mistaken.  This would work, but I am left wondering what reason there was for breaking down this functionality like this.

Reply:
This was my way of only downloading the files actually specified in the SVM poller file for processing.  I could no longer remember if we processed the FLC to generate the SVM FLC and then added the WCS solution to the SVM FLT, or if we do not produce (or no longer produce) an SVM FLT.  Looking at Matt's latest processing, it looks like we do not produce an SVM FLT when CTE processing is done.

Reply:
Makes sense now.  Thanks for the explanation.